app/Main.hs

```diff
@@ -42,10 +42,17 @@ getPlayer playerNumber gameState = case playerNumber of
                                         Player1 -> player1 gameState
                                         Player2 -> player2 gameState 
 
+flagPrompt :: IO Int
+flagPrompt = do
+    input <- ask "Please pick a flag"
+    let result = reads input :: [(Int, String)]
```

Comment:
I'm not sure if I know how that supposed to look like.
Do you mean composing `reads` and `toEnum` like `toEnum . reads` or sth different?